packages/components/src/TooltipTrigger/TooltipTrigger.component.js

```diff
@@ -137,44 +142,96 @@ function TooltipTrigger(props) {
 		};
 	}
 
+	/**
+	 * Activate the tooltip when children are focused
+	 */
+	const onFocus = (...args) => {
+		show();
+		if (props.children.props.onFocus) {
+			props.children.props.onFocus(...args);
+		}
+	};
+
+	/**
+	 * Desactive the tooltip when children are not focused anymore
+	 */
+	const onBlur = (...args) => {
+		hide();
+		if (props.children.props.onBlur) {
+			props.children.props.onBlur(...args);
+		}
+	};
+
+	const onKeyPress = (...args) => {
+		hide();
+		if (props.children.props.onKeyPress) {
+			props.children.props.onKeyPress(...args);
+		}
+	};
+
+	const onMouseOver = (...args) => {
+		show();
+		if (props.children.props.onMouseOver) {
+			props.children.props.onMouseOver(...args);
+		}
+	};
+
+	const onMouseOut = (...args) => {
+		hide();
+		if (props.children.props.onMouseOut) {
+			props.children.props.onMouseOut(...args);
+		}
+	};
+
+	const onClick = (...args) => {
+		hide();
+		if (props.children.props.onClick) {
+			props.children.props.onClick(...args);
+		}
+	};
+
 	const { placement, style } = getTooltipPosition();
 
 	return (
 		// we use div here to wrap tooltip trigger
 		// it should not be reachable
 		// It is just a way to handle click and keyboard events
 		// eslint-disable-next-line jsx-a11y/no-static-element-interactions
-		<div
-			{...omit(props, Object.keys(TooltipTrigger.propTypes))}
-			className={classNames(theme['tc-tooltip'], props.className)}
-			onFocus={show}
-			onBlur={hide}
-			onKeyPress={hide}
-			onMouseOver={show}
-			onMouseOut={hide}
-			onClick={hide}
-			aria-describedby={id}
-			ref={refContainer}
-		>
+		<React.Fragment>
 			{React.Children.map(props.children, child =>
 				cloneElement(child, {
 					'aria-describedby': id,
+					onFocus,
+					onBlur,
+					onKeyPress,
+					onMouseOver,
+					onMouseOut,
+					onClick,
+					ref: node => {
+						refContainer.current = node;
+						const { ref } = child;
+						if (typeof ref === 'function') ref(node);
+						else if (ref) ref.current = node;
+					},
 				}),
 			)}
 
 			{visible &&
 				ReactDOM.createPortal(
-					<div className={theme['tc-tooltip-container']} style={style}>
+					<div className={theme['tc-tooltip-container']} style={getTooltipPosition().style}>
```

Comment:
Maybe directly use `style={style}` ?
You already execute the function on line 193 (`const { placement, style } = getTooltipPosition();`).